sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -148,6 +149,9 @@ class CSVOptions(
     format.setDelimiter(delimiter)
     format.setQuote(quote)
     format.setQuoteEscape(escape)
+    if (charToEscapeQuoteEscaping != '\u0000') {
```

Comment:
Because we always call `format.setQuoteEscape(escape)`, this default value becomes wrong. That means, if users set `\u0000`, the actual quoteEscape in `univocity` is `\`. 

Let us use `Option[Char]` as the type of `charToEscapeQuoteEscaping `? We call `format.setCharToEscapeQuoteEscaping ` only if it is defined.